ncm-systemd/src/main/pan/components/systemd/schema.pan

```diff
@@ -4,12 +4,268 @@
 
 declaration template components/${project.artifactId}/schema;
 
-include 'quattor/schema';
+include 'quattor/types/component';
+include 'components/accounts/functions';
+
+# TODO: some generic types, candidates for template-library-core
+@documentation{
+    hwloc (Portable Hardware Locality, hwloc(7)) location, e.g. node:1 for NUMAnode 1
+}
+type hwloc_location = string with match(SELF, '^[\w:.]+$');
+
+@documentation{
+    syslog facility to use when logging to syslog
+}
+type syslog_facility = string with match(SELF, '^(kern|user|mail|daemon|auth|syslog|lpr|news|uucp|cron|authpriv|ftp|local[0-7])$');
+
+@documentation{
+    syslog level to use when logging to syslog or the kernel log buffer
+}
+type syslog_level = string with match(SELF, '^(emerg|alert|crit|err|warning|notice|info|debug)$');
 
 type ${project.artifactId}_skip = {
     "service" : boolean = false
 };
 
+type ${project.artifactId}_unit_architecture = string with match(SELF, '^(native|x86(-64)?|ppc(64)?(-le)?|ia64|parisc(64)?|s390x?|sparc(64)?)|mips(-le)?|alpha|arm(64)?(-be)?|sh(64)?|m86k|tilegx|cris$');
+
+type ${project.artifactId}_unit_security = string with match(SELF, '^!?(selinux|apparmor|ima|smack|audit)$');
+
+type ${project.artifactId}_unit_virtualization = string with match(SELF, '^(0|1|vm|container|qemu|kvm|zvm|vmware|microsoft|oracle|xen|bochs|uml|openvz|lxc(-libvirt)?|systemd-nspawn|docker)$');
+
+# TODO: make this more finegrained, e.g. has to be existing unit; or check types
```

Comment:
https://github.com/quattor/configuration-modules-core/issues/646, not that trivial to do.